tools/sign_executable/sign_executable.py

```diff
@@ -43,103 +40,48 @@
-            "--username", args.apple_username, "--password", args.apple_password, "--file", notarization_zip]
+            "sudo", "codesign", "-f", "-s", "-v", "--options", "runtime", "--timestamp", "--entitlements", 
+            args.entitlements_file, "--sign", f"Developer ID Application: {args.developer_id}", file]
         spear.log(f"Executing: {' '.join(cmd)}")
-        ps = subprocess.Popen(cmd, stdout=subprocess.PIPE, text=True)
-        request_uuid = ""
-        for line in ps.stdout:
-            spear.log(f"{line}")
-            if "RequestUUID = " in line:
-                request_uuid = line.split("RequestUUID = ")[1].strip()
-        ps.wait()
-        ps.stdout.close()
-        assert request_uuid != ""
-        spear.log(f"Zip file sent for notarization. Request UUID: {request_uuid}")
-    else:
-        request_uuid = args.request_uuid
-
-    # check notarization status
-    cmd = ["xcrun", "altool", "--notarization-info", request_uuid, "--username", args.apple_username, "--password", args.apple_password]
+        subprocess.run(cmd, check=True)
+
+    # create a zip file for notarization
+    notarization_zip = os.path.realpath(os.path.join(args.temp_dir, f"{os.path.splitext(executable_name)[0]}.zip"))
+    cmd = ["ditto", "-c", "-k", "--rsrc", "--keepParent", executable, notarization_zip]
     spear.log(f"Executing: {' '.join(cmd)}")
-    output = "in progress"
-    start_time = time.time()
-    elapsed_time = time.time() - start_time
-    while output == "in progress" and elapsed_time < args.wait_time_seconds:
-        ps = subprocess.Popen(cmd, stdout=subprocess.PIPE, text=True)
-        for line in ps.stdout:
-            spear.log(f"{line}")
-            if "Status:" in line:
-                output = line.split("     Status:")[1].strip()
-        ps.wait()
-        ps.stdout.close()
-        elapsed_time = time.time() - start_time
-        spear.log(f'Waiting to get more information on the notarization request UUID {request_uuid}, current status is "{output}"...')
-    
-    if elapsed_time > args.wait_time_seconds:
-        spear.log(f"Exceeded maximum wait time ({args.wait_time_seconds}s) for request UUID {request_uuid}. Please complete the rest of the procedure after notarization is complete.")
-        assert False
-            
+    subprocess.run(cmd, check=True)
+
+    # send the file for notarization
+    cmd = [
+        "xcrun", "notarytool", "submit", notarization_zip, "--apple-id", args.apple_id,
+        "--team-id", args.apple_teamid, "--password", args.apple_password, "--wait"
+    ]
+    spear.log(f"Executing: {' '.join(cmd)}")
+    ps = subprocess.Popen(cmd, stdout=subprocess.PIPE, text=True)
+    status = ""
+    for line in ps.stdout:
+        spear.log(f"{line}")
+        if "  status: " in line:
+            status = line.split("  status: ")[1].strip()
+    ps.wait()
+    ps.stdout.close()
+    assert status == "Accepted"
```

Comment:
I remember recently you did an extra debugging step where you ran a command-line tool, and you were able to examine a log from the notarization attempt. We should do this extra step after executing `notarytool`, write the log file to `temp_dir` and print it to the console, and then afterwards `assert` that status is accepted.

I'm not set up to run `sign_executable.py`, but if I were, it is a brittle enough procedure that I think I would always want to review the log, even during successful runs.

Reply:
I was on the edge about adding this step. I think writing this log all the time is better because even when the status is successful, there might be some warnings logged in this log file, that is worth looking at (https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow)

Reply:
Added the command to download the log.